### ISSUE TITLE
Support for graphic subtitle formats in cutscenes

### ISSUE THREAD

**Issue by NovachenFS2** [scp-fs2open/fs2open.github.com#1753]
Well, because of the beginning discussion in the Forums: [url=https://www.hard-light.net/forums/index.php?topic=94802.msg1869064;topicseen#msg1869064]Adding subtitles to FS2 cutscenes [/url] and the up- and downsides of SRT files, i create this issue here, because i made my reasons why SRT is not a desireable format for me right now.

So i simply want to see support for subtitles that are based on a graphical layer, as far as i know SUB/IDX (also used on DVDs back then) and PGS (also used on BluRays today) are the most common ones these days. Even they are hard to create, because they are graphic files, they surely would make downloads much smaller without to download whole language specific cutscenes (and yes, i see the downsides of this as well). And you will not have any limitations in the creation of a subtitle, because it is a layer above the video track you can design like you want.

Because i am maybe the only one who would ever use this feature i actually do not see that this will be implemented (which i also can understand ^^). But i wanted to bring it up at least (once again after a discussion in an older topic as well)

**Comment by The-E** [comment]
Does ffmpeg support these formats?

**Comment by NovachenFS2** [comment]
As far as i know, based on this list: [https://www.ffmpeg.org/general.html#File-Formats](url). It should.

**Comment by asarium** [comment]
FFmpeg has support for PGS and DVD subitles but I'll need an example of a movie file with those subtitles before I can look into how this could be supported.

**Comment by NovachenFS2** [comment]
I admit, that i actually never created one myself. But i will try to make one. Could take a while however, because i have to read in in the creation process of that myself.

**Comment by asarium** [comment]
It doesn't have to be a full cutscene. Just a few example subtitles with varying styles would be enough to implement the basic support (or to realize that it's too much work).

**Comment by NovachenFS2** [comment]
Yeah, i actually tried a bit and it seems, that the idea i had in mind is reached by the ASS format actually. And that is a text based subtitle format (never know that), because it allows to reposition, to color and to select invidual fonts for each subtitle and is able to include the needed font file as well.

In the attachment is a small 15 second sequence of FS1' hallfight, but in mp4 of course. This one has two different styles for subtitles. Descriptions and the normal chatter.
The corresponding font was embedded in the ass file.

As for SUB and PGS, actually i do not find a good tutorial to make one so far. And all converters i have found get rid of the graphic style by fonts etc. (that is the reason why i suggest them in the first place)

[hallfight short.zip](https://github.com/scp-fs2open/fs2open.github.com/files/2086982/hallfight.short.zip)
